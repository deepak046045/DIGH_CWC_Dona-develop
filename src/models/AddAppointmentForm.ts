import { AddAppointmentFormFields } from '@/shared/enums/AddAppointmentFormFields';
import { AppointmentTypes } from '@/shared/enums/AppointmentTypes';

export type AddAppointmentForm = {
  [AddAppointmentFormFields.APPOINTMENT_TYPE]: AppointmentTypes | '';
  [AddAppointmentFormFields.INFUSION_NUMBER]: string;
  [AddAppointmentFormFields.ADDRESS]: string;
  [AddAppointmentFormFields.SCHEDULE_DATE]: string;
  [AddAppointmentFormFields.SCHEDULE_TIME]: number;
  [AddAppointmentFormFields.CARE_PARTNERS]: string[];
};
