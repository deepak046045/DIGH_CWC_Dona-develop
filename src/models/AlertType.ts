export type AlertType = {
  isOpen: boolean;
  message: string;
  level: string;
};
