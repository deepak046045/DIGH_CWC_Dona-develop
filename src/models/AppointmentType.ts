import { AppointmentTypes } from '@/shared/enums/AppointmentTypes';

export type AppointmentType = {
  name: AppointmentTypes;
  count?: number;
};

export const APPOINTMENT_ICONS: { [key in AppointmentTypes]: string } = {
  [AppointmentTypes.INFUSION]: '/icons/iv-infusion-icon.svg',
  [AppointmentTypes.GENERAL_DOCTOR]:
    '/icons/general-practicioner-check-in-icon.svg',
  [AppointmentTypes.NEUROLOGIST]: '/icons/neurologist-icon.svg',
  [AppointmentTypes.SAFETY_MRI]: '/icons/safety-mri-icon.svg',
};
