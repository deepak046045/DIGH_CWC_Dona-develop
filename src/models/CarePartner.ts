export type CarePartner = {
  id: string;
  name: string;
  email: string;
};
