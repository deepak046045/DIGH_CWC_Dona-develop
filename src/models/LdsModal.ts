export type LdsModalType = {
  isModalOpen: boolean;
  setIsModalOpen: (arg: boolean) => void;
};
