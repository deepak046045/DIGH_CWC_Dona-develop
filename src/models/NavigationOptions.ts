export type NavigationOption = {
  title: string;
  path: string;
};

export const NavigationOptions: Record<string, NavigationOption> = {
  TreatmentMap: {
    title: 'Treatment map',
    path: '/treatmentMap',
  },
  WhatToExpect: {
    title: 'What to expect',
    path: '/whatToExpect',
  },
  MyTreatmentPlan: {
    title: 'My treatment plan',
    path: '/myTreatmentPlan',
  },
  MyAppointments: {
    title: 'My appointments',
    path: '/myAppointments',
  },
  AddAppointment: {
    title: 'Add appointment',
    path: '/addAppointment',
  },
  MyCarePartners: {
    title: 'My care partners',
    path: '/myCarePartners',
  },
  MyProfile: {
    title: 'My profile',
    path: '/myProfile',
  },
};

export const gatedNavigationOptions: NavigationOption[] = [
  NavigationOptions.MyTreatmentPlan,
  NavigationOptions.MyAppointments,
  NavigationOptions.MyCarePartners,
  NavigationOptions.WhatToExpect,
];

export const ungatedNavigationOptions: NavigationOption[] = [
  NavigationOptions.TreatmentMap,
  NavigationOptions.WhatToExpect,
];
