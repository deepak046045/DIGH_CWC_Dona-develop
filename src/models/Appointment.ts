import { AppointmentType } from './AppointmentType';
import { CarePartner } from './CarePartner';

export type Appointment = {
  id: string;
  date: Date;
  address: string;
  type: AppointmentType;
  carePartners: CarePartner[];
};
