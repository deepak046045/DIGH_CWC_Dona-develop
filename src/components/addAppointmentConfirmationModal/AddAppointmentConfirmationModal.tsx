import React from 'react';
import { LdsModal, LdsButton, LdsIcon } from '@elilillyco/ux-lds-react';
import modal from '@/scss/themes/modal.module.scss';
import buttons from '@/scss/themes/buttons.module.scss';
import { copyTexts } from '@/constants/copyTexts';
import { useRouter } from 'next/navigation';
import { NavigationOptions } from '@/models/NavigationOptions';
import styles from './AddAppointmentConfirmationModal.module.scss';

type AddAppointmentConfirmationModalProps = {
  isModalOpen: boolean;
  setIsModalOpen: (arg: boolean) => void;
  handleAddNewAppointment: () => void;
};

export function AddAppointmentConfirmationModal({
  isModalOpen,
  setIsModalOpen,
  handleAddNewAppointment,
}: AddAppointmentConfirmationModalProps) {
  const router = useRouter();

  return (
    <div className={`${modal.primary} ${styles.modal}`}>
      <LdsModal
        modalId="addCarePartnerModal"
        open={isModalOpen}
        setModalOpen={setIsModalOpen}
        persistent
      >
        <div className={styles.closeModalButtonWrapper}>
          <LdsButton
            onClick={() => {
              router.push(NavigationOptions.MyAppointments.path);
            }}
            clearDefaultClasses
            data-testid="add-appointment-confirmation-x-button"
            className={styles.closeModalButton}
          >
            <LdsIcon name="X" className={styles.closeIcon} />
          </LdsButton>
        </div>
        <h2 className={modal.modalTitle}>
          {copyTexts.AddAppointment.confirmationModal.title}
        </h2>
        <div className={modal.modalActions}>
          <LdsButton
            classes={`${buttons.secondary}`}
            onClick={handleAddNewAppointment}
            data-testid="add-new-appointment-btn"
          >
            {copyTexts.AddAppointment.confirmationModal.secondaryButtonLabel}
          </LdsButton>
          <LdsButton
            classes={`${buttons.primary}`}
            onClick={() => {
              router.push(NavigationOptions.MyAppointments.path);
            }}
            data-testid="view-appointments-btn"
          >
            {copyTexts.AddAppointment.confirmationModal.primaryButtonLabel}
          </LdsButton>
        </div>
      </LdsModal>
    </div>
  );
}
