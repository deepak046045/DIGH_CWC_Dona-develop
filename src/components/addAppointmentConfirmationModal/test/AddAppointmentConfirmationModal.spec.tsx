import { render, screen, fireEvent } from '@testing-library/react';
import React from 'react';
import '@testing-library/jest-dom';
import { useRouter } from 'next/navigation';
import { NavigationOptions } from '@/models/NavigationOptions';
import { AddAppointmentConfirmationModal } from '../AddAppointmentConfirmationModal';

jest.mock('crypto', () => ({ randomUUID: () => '123456789' }));
jest.mock('next/navigation', () => ({
  useRouter: jest.fn(() => ({ push: jest.fn() })),
}));

describe('AddAppointmentConfirmationModal', () => {
  it('should render', () => {
    const { container } = render(
      <AddAppointmentConfirmationModal
        isModalOpen
        setIsModalOpen={() => {}}
        handleAddNewAppointment={() => {}}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('should redirect to my appointments when view all appointments clicked', () => {
    const pushMock = jest.fn();
    (useRouter as jest.Mock).mockReturnValue({
      query: {},
      push: pushMock,
    });

    render(
      <AddAppointmentConfirmationModal
        isModalOpen
        setIsModalOpen={() => {}}
        handleAddNewAppointment={() => {}}
      />
    );

    const viewAppointmentsButton = screen.getByTestId('view-appointments-btn');
    fireEvent.click(viewAppointmentsButton);

    expect(pushMock).toHaveBeenCalledWith(
      NavigationOptions.MyAppointments.path
    );
  });

  it('should redirect to my appointments when close clicked', () => {
    const pushMock = jest.fn();
    (useRouter as jest.Mock).mockReturnValue({
      query: {},
      push: pushMock,
    });

    render(
      <AddAppointmentConfirmationModal
        isModalOpen
        setIsModalOpen={() => {}}
        handleAddNewAppointment={() => {}}
      />
    );

    const closeButton = screen.getByTestId(
      'add-appointment-confirmation-x-button'
    );
    fireEvent.click(closeButton);

    expect(pushMock).toHaveBeenCalledWith(
      NavigationOptions.MyAppointments.path
    );
  });
});
