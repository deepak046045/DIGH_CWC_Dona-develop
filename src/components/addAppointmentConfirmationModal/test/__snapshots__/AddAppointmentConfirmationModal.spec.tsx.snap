// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddAppointmentConfirmationModal should render 1`] = `
<div>
  <div
    class="primary modal"
  >
    <div
      class="lds-modal open"
      id="addCarePartnerModal"
    >
      <div
        class="wrapper"
      >
        <div
          class="row justify-content-center"
        >
          <div
            class="col-12 col-md-9"
          >
            <div
              aria-labelledby=""
              aria-modal="true"
              class="lds-modal-window"
              role="dialog"
            >
              <div
                class="lds-modal-body"
              >
                <div
                  class="closeModalButtonWrapper"
                >
                  <button
                    class="closeModalButton"
                    data-testid="add-appointment-confirmation-x-button"
                    type="button"
                  >
                    <svg
                      aria-labelledby="lds-icon-X-title-123456789 lds-icon-X-description-123456789"
                      class="lds-icon icon-X closeIcon"
                      focusable="false"
                      role="img"
                    >
                      <title
                        id="lds-icon-X-title-123456789"
                      >
                        X
                      </title>
                      <desc
                        id="lds-icon-X-description-123456789"
                      />
                      <svg
                        viewBox="0 0 256 256"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M205.66,194.34a8,8,0,0,1-11.32,11.32L128,139.31,61.66,205.66a8,8,0,0,1-11.32-11.32L116.69,128,50.34,61.66A8,8,0,0,1,61.66,50.34L128,116.69l66.34-66.35a8,8,0,0,1,11.32,11.32L139.31,128Z"
                        />
                      </svg>
                    </svg>
                  </button>
                </div>
                <h2
                  class="modalTitle"
                >
                  Your appointment has been saved.
                </h2>
                <div
                  class="modalActions"
                >
                  <button
                    class="lds-button secondary"
                    data-testid="add-new-appointment-btn"
                    type="button"
                  >
                    Add new appointment
                  </button>
                  <button
                    class="lds-button primary"
                    data-testid="view-appointments-btn"
                    type="button"
                  >
                    View my appointments
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
